### PR TITLE
Report multibuild flavors if package is multibuild

### PR DESCRIPTION
**Pull request by vpereira**
Report to SCM the multibuild flavor if target package is multibuild

### CODE REVIEW

File: src/api/app/models/workflow/step.rb [hunk 1]
```diff
@@ -156,9 +168,11 @@ def report_to_scm(workflow_filters)
     workflow_repositories(target_project_name, workflow_filters).each do |repository|
       # TODO: Fix n+1 queries
       workflow_architectures(repository, workflow_filters).each do |architecture|
-        # We cannot report multibuild flavors here... so they will be missing from the initial report
-        SCMStatusReporter.new({ project: target_project_name, package: target_package_name, repository: repository.name, arch: architecture.name },
-                              scm_webhook.payload.merge({ short_package_name: target_package_name(short_commit_sha: true) }), @token.scm_token).call
+        target_package_names.each do |target_package_name_or_flavor|
+          # report multibuild flavors here...
```
Comment (dmarcoux) [inline src/api/app/models/workflow/step.rb:172]: I would remove this comment, it doesn't provide much information.

```suggestion
```

File: src/api/app/models/workflow/step.rb [hunk 2]
```diff
@@ -88,6 +92,14 @@ def target_package_name(short_commit_sha: false)
 
   private
 
+  def multibuild?
+    Package.get_by_project_and_name(target_project_name, target_package_name, follow_multibuild: true).multibuild?
+  end
+
+  def multibuild_flavors
+    Package.get_by_project_and_name(target_project_name, target_package_name, follow_multibuild: true).multibuild_flavors.collect { |_flavor| "#{target_package_name}:flavor" }
```
Comment (dmarcoux) [inline src/api/app/models/workflow/step.rb:100]: We need `flavor`, so it shouldn't be prefixed with `_`.

```suggestion
    Package.get_by_project_and_name(target_project_name, target_package_name, follow_multibuild: true).multibuild_flavors.collect { |flavor| "#{target_package_name}:flavor" }
```

File: src/api/app/models/workflow/step.rb [hunk 3]
```diff
@@ -88,6 +92,14 @@ def target_package_name(short_commit_sha: false)
 
   private
 
+  def multibuild?
```
Comment (dmarcoux) [inline src/api/app/models/workflow/step.rb:95]: This method isn't needed anymore. It can be removed.

**Comment by hennevogel** [comment]
[Initial reporting to SCM isn't working for flavors of multibuild packages](https://trello.com/c/3L5iAioF/1817-initial-reporting-to-scm-isnt-working-for-flavors-of-multibuild-packages)

**Comment by hennevogel** [comment]
can you please seperate the cassette changes from the code changes to make it easier to review?

File: src/api/app/models/workflow/step.rb [hunk 4]
```diff
@@ -43,7 +46,7 @@ def create_or_update_subscriptions(package, workflow_filters)
                                                           enabled: true,
                                                           token: @token,
                                                           package: package)
-      subscription.update!(payload: scm_webhook.payload.merge({ workflow_filters: workflow_filters, short_package_name: target_package_name(short_commit_sha: true) }))
```
Comment (dmarcoux) [inline src/api/app/models/workflow/step.rb:46]: We don't use `short_package_name` anymore in `ReportToScmJob` where this payload is used, so this why we remove this key from the payload.

File: src/api/spec/models/workflow/step/branch_package_step_spec.rb [hunk 5]
```diff
@@ -223,6 +219,57 @@
         it_behaves_like 'fails with insufficient write permission on target project'
       end
 
+      context 'for a multibuild package' do
```
Comment (dmarcoux) [inline src/api/spec/models/workflow/step/branch_package_step_spec.rb]: This is mostly a copy-and-paste from the `'successful new PR or MR event'` shared context in this file. Shared contexts are a nightmare to maintain and on top of this, it didn't pass for a muiltibuild package, so it had to be changed.

File: src/api/spec/models/workflow/step/branch_package_step_spec.rb [hunk 6]
```diff
@@ -272,8 +319,7 @@
 
         let(:octokit_client) { instance_double(Octokit::Client) }
         let(:target_project_final_name) { "home:#{user.login}" }
-        let(:final_package_name) { "#{package.name}-#{long_commit_sha}" }
-        let(:final_short_package_name) { "#{package.name}-#{short_commit_sha}" }
+        let(:final_package_name) { "#{package.name}-#{short_commit_sha}" }
```
Comment (dmarcoux) [inline src/api/spec/models/workflow/step/branch_package_step_spec.rb]: There wasn't a short and long package name, this was wrong.

File: src/api/spec/models/workflow/step/branch_package_step_spec.rb (hunk 5 above)
Comment (dmarcoux) [inline src/api/spec/models/workflow/step/branch_package_step_spec.rb]: This is mostly a copy-and-paste from the `'successful new PR or MR event'` shared context in this file. Shared contexts are a nightmare to maintain and on top of this, it didn't pass for a muiltibuild package, so it had to be changed.

**Review by krauselukas — APPROVED** [review]
Tested it locally and it worked :) code wise it looks good to me as well! Nice @vpereira @dmarcoux

**Pull request merged**